resources/js/utils/throttle.ts

```diff
@@ -0,0 +1,25 @@
+/**
+ * Creates a throttled version of a function that can be called at most
+ * once per `waitMs` milliseconds.
```

Comment:
Could this be lodash's debounce instead of a custom implementation?

Reply:
`lodash.throttle` might achieve the same thing - I will check. It's a subtle difference from `debounce` because we are not trying to delay the call, only rate limit it.

Reply:
I have kept the wrapper in case we want to swap this dep at some point in the future (happy to remove if you'd prefer). `lodash.throttle` works fine here.